ThunderCatsTests/UIColorTests.m

```diff
@@ -81,6 +81,49 @@ - (void)testColorWithHexString
     XCTAssertFalse(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
 }
 
+- (void)testColorWithShortHexString
+{
+    UIColor *color1 = [UIColor tc_colorWithHexString:@"000"];
+    UIColor *color2 = [UIColor blackColor];
+    
+    CGFloat r1, r2, g1, g2, b1, b2, a1, a2;
+    [color1 getRed:&r1 green:&g1 blue:&b1 alpha:&a1];
+    [color2 getRed:&r2 green:&g2 blue:&b2 alpha:&a2];
+    
+    XCTAssertTrue(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
```

Comment:
This assertion isn't good enough.  It can pass for any number of values, really.  And problematically, it can fail for too many reasons.

It could fail because any one of six values was returned incorrectly.

Instead, we should write tests that look more like this...

`UIColor *black = [UIColor tc_colorWithHexString:@"000"];
CGFloat blackRed, blackGreen, blackBlue, blackAlpha;
[black getRed:&blackRed green:&blackGreen blue:&blackBlue alpha:&blackAlpha];

XCTAssertEqualWithAccuracy(blackRed, 0.0, 0.05, @"Expected red value of 0, but got %f", blackRed);
XCTAssertEqualWithAccuracy(blackGreen, 0.0, 0.05, @"Expected green value of 0, but got %f", blackGreen);
XCTAssertEqualWithAccuracy(blackBlue, 0.0, 0.05, @"Expected blue value of 0, but got %f", blackBlue);
XCTAssertEqualWithAccuracy(blackAlpha, 1.0, 0.05, @"Expected alpha value of 0, but got %f", blackAlpha);`

- Each assertion tests exactly one thing.  
- Each assertion compares an actual output to an expected output.
- Each assertion has a clear message that specifies what didn't pass _exactly_.